dbt/adapters/snowflake/impl.py

```diff
@@ -135,7 +135,9 @@ def list_relations_without_caching(self, schema_relation: SnowflakeRelation) ->
             # if the schema doesn't exist, we just want to return.
             # Alternatively, we could query the list of schemas before we start
             # and skip listing the missing ones, which sounds expensive.
-            if "Object does not exist" in str(exc):
+            # "002043 (02000)" is error code for "object does not exist or is not found"
+            # The error messages are different in English and Japanese, need each handling.
```

Comment:
This solution is applicable for languages other than Japanese and English, so it would better to either remove this comment or update it to be more generic.

Here's what a more generic version could look like:

```suggestion
            # The error message text may vary across languages, but the error code is expected to be more stable
```